crates/oxc_linter/src/rules/react/display_name.rs

```diff
@@ -0,0 +1,2402 @@
+use schemars::JsonSchema;
+use serde::Deserialize;
+
+use oxc_ast::{
+    AstKind,
+    ast::{
+        AssignmentTarget, ClassElement, Expression, ObjectExpression, ObjectPropertyKind,
+        PropertyKey, Statement,
+    },
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_ecmascript::PropName;
+use oxc_macros::declare_oxc_lint;
+use oxc_semantic::{AstNode, Reference, SymbolId};
+use oxc_span::{CompactStr, GetSpan, Span};
+
+use crate::{
+    context::LintContext,
+    rule::{DefaultRuleConfig, Rule},
+    utils::{
+        contains_jsx, find_innermost_function_with_jsx, function_contains_jsx, is_hoc_call,
+        is_react_component_name,
+    },
+};
+
+fn component_display_name_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Component definition is missing display name")
+        .with_help("Add a displayName property to the component")
+        .with_label(span)
+}
+
+fn context_display_name_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Context definition is missing display name")
+        .with_help("Add a displayName property to the context")
+        .with_label(span)
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that React components have a displayName property.
+    ///
+    /// ### Why is this bad?
+    /// React DevTools uses displayName to show component names in the component tree.
+    /// Without displayName, components will show up as "Unknown" in DevTools.
+    ///
+    /// ### Example
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```jsx
+    /// const MyComponent = () => <div>Hello</div>;
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```jsx
+    /// const MyComponent = () => <div>Hello</div>;
+    /// MyComponent.displayName = 'MyComponent';
+    /// ```
+    DisplayName,
+    react,
+    style,
+    config = DisplayNameConfig,
+);
+
+#[derive(Debug, Clone)]
+enum ComponentType {
+    Function,
+    Class,
+    CreateReactClass,
+    ObjectProperty,
+}
+
+#[derive(Debug)]
+struct ReactComponentInfo {
+    span: Span,
+    _component_type: ComponentType,
```

Comment:
Any reason we need the underscore here?


```suggestion
    component_type: ComponentType,
```

Reply:
It's unused code. I'm guessing it's prefixed with an `_` so that the compiler doesn't report a dead code warning.

We should just remove this struct property.